Assignment6.Rmd

```diff
@@ -1,86 +1,180 @@
+
+(Question: what does it mean when there are multiple variables, but only 1 is used in tree construction? Understand it means cp does not reduce lack of fit by 0.01 with the next split, but does that mean it's the most important variable?)
+
+```{r own df}
+student_course <- read.csv("student.course.csv")
+student_record <- read.csv("student.record.csv")
 
+View(student_course)
+View(student_record)
+
+student_course1 <- student_course %>%
+  select(ANONID, GRD_PTS_PER_UNIT)
+
+#Note: I initially included `GPAO`, but when I do, all other variables are not used. Only `GPAO` was used in the construction of the tree.
+
+student_record1 <- student_record %>%
+  select(ANONID, MAJOR1_DESCR, HSGPA, LAST_ACT_ENGL_SCORE, LAST_ACT_MATH_SCORE, LAST_ACT_READ_SCORE, LAST_ACT_SCIRE_SCORE, LAST_ACT_COMP_SCORE, LAST_SATI_VERB_SCORE, LAST_SATI_MATH_SCORE, LAST_SATI_TOTAL_SCORE) %>%
+  na.omit()
+
+df <- student_course1 %>%
+  left_join(student_record1,by="ANONID") %>%
+  na.omit()
+#Note: df does not have unique ANONID, as one student might take several courses within PLA? 
+
+df$MAJOR_BS <- str_detect(df$MAJOR1_DESCR, "BS")
+
+df <- df %>%
+  select(-MAJOR1_DESCR) %>%
+  select(-ANONID)
+
+```
+
+```{r model with variables}
+
+rpart3<- rpart(GRD_PTS_PER_UNIT ~ ., method = "anova", data = df)
+
+printcp(rpart3)
+rpart.plot(rpart3)
 ```
 
+```{r model with PCA}
+
+df2 <- df %>%
+  select(-GRD_PTS_PER_UNIT)
+
+df_scaled <- scale(df2, center = TRUE)
+pca <- prcomp(df_scaled, scale = TRUE)
+
+pca
+summary(pca)
+
+df_pca <- as.data.frame(pca$x) %>%
+  select(-c(PC7, PC8, PC9, PC10))
+
+df_pca <- cbind(df_pca, df$GRD_PTS_PER_UNIT)
+
+rpart4 <- rpart(df$GRD_PTS_PER_UNIT ~ ., method = "anova", data = df_pca)
+
+printcp(rpart4)
+
+rpart.plot(rpart4)
+
+```
 
-### To Submit Your Assignment
+The model that uses PCA performs significantly better (xerror = 0.012) as compared to the model based on raw variables (xerror = 0.909), according to the xerror cross validation metric. The PCA may have reduced features that have multicollinearity (likely all the variables with `LAST_ACT_<subj>_SCORE`), resulting in lower prediction power. 
 
-Please submit your assignment by first "knitting" your RMarkdown document into an html file and then commit, push and pull request both the RMarkdown file and the html file.
\ No newline at end of file
+(Question: The following was the result when I initially included `GPAO` (and only `GPAO` is used in tree construction): The model that uses raw variables is slightly better (xerror = 0.753) as compared to the PCA model (xerror = 0.762), according to the xerror cross validation metric. Question: why is that so? Is it because in this case, only `GPAO` is used in the tree construction anyway, and the PCA might have reduced features in order and reduced overfitting, therefore increasing CV error?)
```

Comment:
In theory, feature selection with PCA is supposed to reduce the error rate since it removes the "noise" since it results in features that account for the maximum variance in terms of data. Great job overall! Keep up the good work.

Sources:
https://stats.stackexchange.com/questions/27750/feature-selection-and-cross-validation
https://stats.stackexchange.com/questions/55034/how-does-pca-improve-the-accuracy-of-a-predictive-model